### DISCUSSION TITLE
How I can make a keyboard shortcut to delete a whole word?

### DISCUSSION THREAD

**Discussion by tommasoxd** [kovidgoyal/kitty#3286] [Q&A]
In alacritty when I press ctrl+backspace it deletes the whole word, how I can do this in kitty?

**Comment by Luflosi** [comment]
<kbd>alt</kbd>+<kbd>backspace</kbd> works for me.
I'm using macOS, not sure if that matters.

**Reply by tommasoxd** [reply]
worked thank you much 👍🏻 .

**Reply by bebraw** [reply]
As a related question, how to remap this to cmd? I imagine there must be some way. It's good to know alt+backspace does the trick, though. 👍

**Reply by trygveaa** [reply]
@bebraw: You can use `send_text`: https://sw.kovidgoyal.net/kitty/conf.html?highlight=send_text#shortcut-kitty.Send-arbitrary-text-on-key-presses

**Reply by roadkell** [reply]
@bebraw I'm not a Mac user, but the solution can probably be adapted. Unix control code for backspacing a word is Ctrl-w (^W in caret notation) which in kitty.conf looks like `map ctrl+backspace send_text all \x17`

**Reply by nivethan-me** [reply]
> @bebraw I'm not a Mac user, but the solution can probably be adapted. Unix control code for backspacing a word is Ctrl-w (^W in caret notation) which in kitty.conf looks like `map ctrl+backspace send_text all \x17`

it's working on my ubuntu 20.04 , thank you !

**Reply by tommasoxd** [reply]
thank to all for the replies 😄

**Reply by PabloGmz96** [reply]
> @bebraw I'm not a Mac user, but the solution can probably be adapted. Unix control code for backspacing a word is Ctrl-w (^W in caret notation) which in kitty.conf looks like `map ctrl+backspace send_text all \x17`

Also works for me on Zorin16, thank you very much!

**Reply by alarbada** [reply]
> @bebraw I'm not a Mac user, but the solution can probably be adapted. Unix control code for backspacing a word is Ctrl-w (^W in caret notation) which in kitty.conf looks like map ctrl+backspace send_text all \x17


I've wanted this mapping for like 7 years. I'm sick of using ctrl + w to delete a whole word, as I keep using that same mapping in the browser, making me unvoluntarily close a tab that I did not want to close. Now I can just forget about ctrl + w.
THANK YOU

**Comment by ian-kelling** [comment]
In my bashrc, `stty werase '^h'`, fixes it for me.

**Reply by tommasoxd** [reply]
thanks for the suggestion I’ll try

**Comment by akharrou** [comment]
you can accomplish this at the shell level too, independent of what terminal you use. for example w/ zsh you can write the following in your `.zshrc`: `bindkey <your-trigger-keyseq> backward-delete-word`, then to find what key sequence is trigger by your desired key you can try `$> cat` then type the key, then take that and plug it into `<your-trigger-keyseq>`

**Reply by paoloose** [reply]
thanks! `bindkey "^H" backward-delete-word` works perfectly for ctrl+backspace
